### PR TITLE
TEN-1381 / 24.10 / Add POSIX, NFSv4 and SMB ACL Testing

### PR DESCRIPTION
**Pull request by jordan-mcgillivray005**

### CODE REVIEW

**Comment by jordan-mcgillivray005** [comment]
Test fully functional, failing for NAS-129686 as expected. All the green from the mark is the full test, the 3 recent reds are from the new defect. 
image
image

File: keywords/webui/smb.py [hunk 1]
```diff
@@ -2,11 +2,47 @@
...
+        :param type_field: Optional, defaults to ALLOWED. [ALLOWED/DENIED]
+
+        Example:
+            - SMB.add_additional_acl_who_entry('group', 'admin_group')
+            - SMB.add_additional_acl_who_entry('everyone')
+        """
+        COM.click_button("add-item-add-entry")
+        WebUI.delay(0.1)
+        COM.click_on_element(xpaths.smb.smb_acl_empty_entry_who())
+        match who_type:
```
Comment (markbainix) [inline keywords/webui/smb.py:27]: suggest: adding ".lower()" to ensure lowercase comparison

File: keywords/webui/smb.py [hunk 2]
```diff
@@ -2,11 +2,47 @@
...
+            case 'user':
+                who_type_cap = 'User'
+            case 'group':
+                who_type_cap = 'Group'
+            case 'everyone':
+                who_type_cap = 'everyone'
+        COM.click_on_element(xpaths.common_xpaths.option_field(f"ae-who-{who_type}"))
```
Comment (markbainix) [inline keywords/webui/smb.py:34]: suggest: adding ".lower()" to ensure lowercase comparison

File: keywords/webui/smb.py (hunk 1 above)
Comment (jordan-mcgillivray005) [reply keywords/webui/smb.py:28]: Done

**Review by markbainix — APPROVED** [review]

**Review by markbainix — APPROVED** [review]

**Review by ericbsd — APPROVED** [review]

**Pull request merged**